src/pages/index.js

```diff
@@ -88,38 +220,92 @@ document.querySelector(".profile__add-button").addEventListener("click", () => {
 const imagePopup = new PopupWithImage("#preview-modal");
 imagePopup.setEventListeners();
 
-function handleImageClick(link, name) {
-  imagePopup.open({ name, link });
-}
 /* ------------------------------------------------- */
-/*                 Render Cards/Image 
+/*                 Rendering Cards 
 /* ------------------------------------------------- */
 function createCard(item) {
-  const cardElement = new Card(item, "#card-form", handleImageClick);
+  //console.log("Item data:", item);
+  const cardElement = new Card(
+    item,
+    "#card-form",
+    handleImageClick,
+    handleDeleteClick,
+    confirmPopup,
+    handleLikeCard,
+    handleUnlikeCard
+  );
+
   return cardElement.createCard();
 }
-const renderer = (cardData) => {
-  const cardElement = createCard(cardData);
-  section.addItem(cardElement);
-};
 
-const section = new Section({ items: initialCards, renderer }, ".card__list");
-//console.log(initialCards);
-section.renderItems();
-/* ------------------------------------------------- */
-/*                     Form Validation
-/* ------------------------------------------------- */
-const formValidators = {};
+const confirmPopup = new PopupConfirmation("#delete-confirm-modal");
+confirmPopup.setEventListeners();
 
-const enableValidation = (settings) => {
-  const formList = Array.from(document.querySelectorAll(settings.formSelector));
-  formList.forEach((formElement) => {
-    const validator = new FormValidator(settings, formElement);
-    const formName = formElement.getAttribute("name");
+function handleImageClick(link, name) {
+  //console.log("Preview modal triggered for:", link, name);
+  imagePopup.open({ name, link });
+}
 
-    formValidators[formName] = validator;
-    validator.enableValidation();
-  });
-};
+function handleDeleteClick(cardId, cardElement) {
+  console.log("Deleting card with ID:", cardId);
 
-enableValidation(settings);
+  confirmPopup.renderLoading(true, "Deleting...");
+
+  api
+    .removeCard(cardId)
+    .then(() => {
+      console.log("Card successfully deleted from server");
+      cardElement.remove();
+      confirmPopup.close();
+      console.log("Popup should close now");
+    })
+    .catch((err) => console.error("Failed to delete card:", err))
+    .finally(() => {
+      confirmPopup.renderLoading(false);
+    });
+}
+
+confirmPopup.setSubmitAction(() => {
+  console.log("Yes button clicked - confirming delete");
+  handleDeleteClick(confirmPopup.cardId, confirmPopup.cardElement);
+});
+
+function handleLikeCard(cardId, likeButton) {
+  const isLiked = likeButton.classList.contains("card__like-button_active");
+  const apiCall = isLiked ? api.dislikeCard(cardId) : api.likeCard(cardId);
+  console.log(
+    `Attempting to ${isLiked ? "remove" : "add"} like for card ID: ${cardId}`
+  );
+
+  apiCall
+    .then((updatedData) => {
+      console.log("Server responded successfully:", updatedData);
+      likeButton.classList.toggle("card__like-button_active");
```

Comment:
`      likeButton.classList.toggle("card__like-button_active");
`

Now only class `Card` knows every **card selector**, so you need to encapsulate the likes logics inside its methods and call them in `index.js`. It’s a violation of **Single responsibility** principle